### PR TITLE
Wrap printing of object parsing and muscle negative force warnings with Object::_debugLevel 

### PR DESCRIPTION
**Pull request by nickbianco**
This PR adds a static member variable to the `Muscle` class to allow the user to disable warnings that get printed out when actuation goes negative. We use this in Moco to avoid the output stream from becoming flooded with error messages during an optimization (e.g. we're okay with negative values while calculating sparsity for the Jacobian). 

Output messages in `Storage` and `ExternalForce` are commented out for a similar reason: messages get printed out everytime `initSystem()` is called (which can happen many times while solving a problem in Moco), and this pollutes the output making it hard to find the useful information we print out when solving a problem. It's outside the scope of this PR, but a general scheme for handling logging and when information is or isn't printed out would be appropriate for these messages.

---
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/2541)

### CODE REVIEW

File: OpenSim/Simulation/Model/Muscle.cpp [hunk 1]
```diff
@@ -39,6 +39,8 @@ using SimTK::Vec3;
 
 static const Vec3 DefaultMuscleColor(.8, .1, .1); // Red for backward compatibility
 //static int counter=0;
+bool Muscle::_printWarnings = true;
```
Comment (tkuchida) [inline OpenSim/Simulation/Model/Muscle.cpp]: Could use Object's `_debugLevel` instead?

File: OpenSim/Simulation/Model/Muscle.h [hunk 2]
```diff
@@ -333,6 +333,16 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Muscle, PathActuator);
...
     double computeActuation(const SimTK::State& s) const override = 0;
 
+    /** Whether or not to print warnings (e.g. when muscle force goes negative). 
+        May be useful for optimization to ingore warnings when computing finite
+        differences. **/
+    static void setPrintWarnings(bool tf) {
+        _printWarnings = tf;
+    };
+    /** Get the current setting of the print warning flag. **/
+    static bool getPrintWarnings() {
```
[the remark this answers is not in the document]
Comment (chrisdembia) [reply OpenSim/Simulation/Model/Muscle.h]: Static functions can’t be marked const bc they can’t affect member variables anyway.

**Comment by jenhicks** [comment]
Model and external loads loaded: should be printed at level 0
Muscle negative force: should be printed at level 0
Level -1 (quiet) will not print these

Make sure that there are not any equality statements that should be inequality (to account for the level -1).

**Comment by aymanhab** [comment]
@nickbianco do you mind changing the title/name of the PR to reflect what it actually does? Since it does not add flags to the Muscle class and it affects the behavior of other classes as well. Thank you.

**Comment by aymanhab** [comment]
LGTM, just a couple minor clarifications and this should be good to merge. Thanks @nickbianco

**Comment by chrisdembia** [comment]
___
*[OpenSim/Simulation/Model/Muscle.cpp, line 670 at r1](https://reviewable.io/reviews/opensim-org/opensim-core/2541#-LmQiOsVoxFlyHo-J6Yq-r1-670:-LmvhH9H3f1deCUziP5L:b-sa1nu3) ([raw file](https://github.com/opensim-org/opensim-core/blob/ac8ce8a1735b68fa3e4e6702f6e49ad6398bf44e/OpenSim/Simulation/Model/Muscle.cpp#L670)):*
<details><summary><i>Previously, nickbianco (Nicholas Bianco) wrote…</i></summary><blockquote>

@aymanhab I'm not sure what you mean here. getActuation() and isActuationOverriden() shouldn't be affected by debug level. 
</blockquote></details>

Previously, `isActuationOverriden(s)` and `getActuation(s)` were invoked regardless of debug level. Now, there's a situation where these functions are no longer called. But these functions might be doing something other than returning their values. For example, they could be realizing the state (a side effect), and this would no longer occur, forcing other code outside of this function to change.

**Comment by aymanhab** [comment]
Thanks @nickbianco 👍  I looked into the implementation and indeed it seems to be not (at least explicitly) calling realize. I just wanted us to be vigilant going forward, especially when we introduce logging, that changes to logging do not inadvertently change results.

**Pull request merged**